File: chunyang/hw1/hw1.m
n = [0  25	30	35	40	45	50];
t = [0  0	0.01	0.17	1.97	21.92	242.360001];
figure(1);
plot(n,t);
xlabel('n (recursive algorithm)');
ylabel('Running time (s)');

x = [0  1	10	100	1000	2000	5000	10000	20000	50000].*1e6;
y = [0  0	0.03	0.28	2.78	5.56	13.89	27.809999	55.630001	139.009995];
figure(2);
plot(x,y);
xlabel('n (iterative algorithm)');
ylabel('Running time (s)');

File: chunyang/hw1/hw1.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int recursivefib(int n)
{
	if (n == 0)
		return 0;
	else if (n == 1)
		return 1;
	else
		return (recursivefib(n - 1) + recursivefib(n - 2));
}

long iterativefib(long n)
{
	long fib = 0, a = 1, t = 0;
	for (long k = 0; k < n; k++)
	{
		t = fib + a;
		a = fib;
		fib = t;
	}
	return fib;
}

int main()
{
	clock_t timer;

	/*recursive algorithm*/
	int n = 30;
	int recursivesum;
	timer = clock();
	recursivesum = recursivefib(n);
	timer = clock() - timer;
	printf("The %dth number of this Fibonacci Numbers using recursive algorithm is: %d\n", n, recursivesum);
	printf("recursive algorithm for n = %d ran in %f seconds\n\n", n, (float)timer / (float)CLOCKS_PER_SEC);
	
	/*iterative algorithm*/
	long n = 10000000;
	long iterativesum;
	timer = clock();
	iterativesum = iterativefib(n);
	timer = clock() - timer;
	printf("The %ldth number of this Fibonacci Numbers using iterative algorithm is: %ld\n", n, iterativesum);
	printf("iterative algorithm for n = %ld ran in %f seconds\n\n", n, (float)timer / (float)CLOCKS_PER_SEC);
}